python/pyspark/sql/connect/session.py

```diff
@@ -176,6 +180,27 @@ def enableHiveSupport(self) -> "SparkSession.Builder":
                 error_class="NOT_IMPLEMENTED", message_parameters={"feature": "enableHiveSupport"}
             )
 
+        def _apply_options(self, session: "SparkSession") -> None:
+            with self._lock:
+                for k, v in self._options.items():
+                    try:
+                        session.conf.set(k, v)
+                    except AnalysisException as e:
+                        current = currentframe()
```

Comment:
What are you trying to do here BTW?

Reply:
In the unit tests, the exception that was raised when invalid configuration key was provided was always analysis exception so I used `AnalysisException` in the except.

I was originally trying to provide detailed warnings but after seeing @zhengruifeng's suggestions I see that providing a simple warning with the details of the exception should be ample for users.